pkg/util/tar_utils.go

```diff
@@ -106,21 +106,52 @@ func unpackTar(tr *tar.Reader, path string) error {
 				return err
 			}
 			currFile.Close()
-		}
+		case tar.TypeSymlink:
+			// It's possible we end up creating files that can't be overwritten based on their permissions.
+			// Explicitly delete an existing file before continuing.
+			if _, err := os.Stat(target); !os.IsNotExist(err) {
+				logrus.Debugf("Removing %s to create symlink.", target)
+				if err := walkAndRemove(target); err != nil {
+					logrus.Debugf("Unable to remove %s: %s", target, err)
+				}
+			}
 
+			if err = os.Symlink(header.Linkname, target); err != nil {
+				logrus.Errorf("Failed to create symlink between %s and %s: %s", header.Linkname, target, err)
+			}
+		}
 	}
 	return nil
 }
 
+func walkAndRemove(p string) error {
+	return filepath.Walk(p, func(path string, info os.FileInfo, err error) error {
+		if e := os.Chmod(path, 0777); e != nil {
```

Comment:
Why do we need to chmod before removal?

Reply:
Ah sorry, I was trying something out and meant to remove that